### PR TITLE
Fix typos/broken links; improve code

### PR DESCRIPTION
**Pull request by OnkarRuikar**
The PR pulls following code change:
- update deprecated values
- some trivial code simplification
- updates markdown code fence languages

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ These resources are helpful for updating Chromium-based browsers, including Chro
 - [Safari Release Notes](https://developer.apple.com/documentation/safari-release-notes): A list of release notes for different versions of Safari.
 - [WebKit Bugzilla](https://bugs.webkit.org/): The bug tracker for Safari/WebKit.
 - [WebKit Repo](https://github.com/WebKit/WebKit): The source code repository for WebKit, the engine for Safari. Useful to find specific behavior changes that are hard to test in the browser.
-- [Backported Releases List](issue-triage-checklist.md#backported-releases): A list of Safari releases that were backports of newer versions and should be ignored.
+- [Backported Releases List](./data-guidelines/index.md#backported-releases): A list of Safari releases that were backports of newer versions and should be ignored.
```
Comment (teoli2003) [inline docs/contributing.md:89]: Why the `./`? Wouldn't this work? (Not tested)

```suggestion
- [Backported Releases List](data-guidelines/index.md#backported-releases): A list of Safari releases that were backports of newer versions and should be ignored.
```

**Review by queengooborg — CHANGES_REQUESTED** [review]
Hi @OnkarRuikar -- thanks for your PR and working hard to fix issues within BCD.  There's a lot of different changes to review, however.  Would you mind splitting this into multiple PRs based on the following (and preferably, add PR descriptions for the rationale behind the changes, if applicable)?

- HTTP -> HTTPS
- Typo/grammar fixes
- Link fixes
- Code fixes

File: docs/contributing.md (hunk 1 above)
Comment (OnkarRuikar) [reply docs/contributing.md]: Done. Many places do use `./`, e.g. [the same document](https://github.com/mdn/browser-compat-data/blob/46eb429dcf1e6c5149073d57cf0b1f57d8c3ac43/docs/contributing.md?plain=1#L111). Do we want to update them to too?

Comment (teoli2003) [reply docs/contributing.md]: I don't know if this has an effect I would ignore, so let's wait for this PR to be approved (I will not do it myself as I'm not confident enough here, but LGTM)

**Pull request merged**